pyxem/utils/io_utils.py

```diff
@@ -27,89 +27,110 @@
+def load(filename, lazy=False,
+         is_ElectronDiffraction2D=True,
+         scan_size=None, sum_length=10):
+    """Load data into pyxem objects.
 
-    return dp
-
-def load(filename,is_ElectronDiffraction=True):
-    """
-    A wrapper around hyperspy's load function that enables auto-setting signals to ElectronDiffraction
-    and correct loading of previously saved ElectronDiffraction, TemplateMatchingResults and DiffractionVectors
-    objects
+    Wraps hyperspy.load to load various file formats and assigns suitable
+    loaded data to pyxem signals.
 
     Parameters
     ----------
 
     filename : str
         A single filename of a previously saved pyxem object. Other arguments may
         succeed, but will have fallen back on hyperspy load and warn accordingly
-    is_ElectronDiffraction : bool
+    lazy : bool
+        If True the file will be opened lazily, i.e. without actually reading
+        the data from the disk until required. Allows datasets much larger than
+        available memory to be loaded.
+    is_ElectronDiffraction2D : bool
         If the signal is not a pxm saved signal (eg - it's a .blo file), cast to
-        an ElectronDiffraction object
+        an ElectronDiffraction2D object
+    scan_size : int
+        Scan size in pixels, allows the function to reshape the array into
+        the right shape when loading a .mib file.
+    sum_length : int
+        Number of lines to sum over to determine scan fly back location.
+
+    Returns
+    -------
+    s : Signal
+        A pyxem Signal object containing loaded data.
     """
-    if isinstance(filename,str) == True:
+    if isinstance(filename, str) == True:
         file_suffix = '.' + filename.split('.')[-1]
     else:
         warnings.warn("filename is not a single string, for clarity consider using hs.load()")
-        s = hyperspyload(filename)
+        s = hyperspyload(filename, lazy=lazy)
         return s
 
     if file_suffix == '.mib':
-        raise ValueError('mib files must be loaded directly using pxm.load_mib()') #pragma: no cover
-
-
-    signal_dictionary = {'electron_diffraction':ElectronDiffraction,
-                         'template_matching':TemplateMatchingResults,
-                         'diffraction_vectors':DiffractionVectors,
-                         'crystallographic_map':CrystallographicMap,
-                         'diffraction_profile':ElectronDiffractionProfile,
-                         'vdf_image':VDFImage}
-
-    if file_suffix in ['.hspy','.blo']: # if True we are loading a signal from a format we know
-        s = hyperspyload(filename)
-        try:
-            s = signal_dictionary[s.metadata.Signal.signal_type](s)
-        except KeyError:
-            if is_ElectronDiffraction:
-                s = ElectronDiffraction(s)
-            else:
-                warnings.warn("No pyxem functionality used, for clarity consider using hs.load()")
+        dpt = load_with_reader(filename=filename, reader=mib_reader)
```

Comment:
this hard codes a number of args that existed in previous function (which is why it wasn't wrapped originally)

Reply:
Why? the previous function was:  def load_mib(filename, scan_size, sum_length=10) and I added those args as kwargs to the load function?

Reply:
So you have, mea culpa. Although the default scan_size argument will throw an error now.

Reply:
I also think the `load_mib` function should at very least still be separate and called from `load` - for modularity and readability reasons if nothing else (you can add a leading underscore if you want) & I think what we had was better than this is

Reply:
Ok - I think I've flipped on this towards even more granularity i.e.: load, load_mib, load_hspy, which will cover:

load - pyXem saved files only.

load_hspy - for hyperspy supported EM file formats most of which may contain ElectronDiffraction data. This should have an "assign_to" argument replacing the current "is_ElectronDiffraction" so that the user can specify what pyxem signal to put it in if they want to.

load_mib - for loading .hdr/.mib pairs. I have a lazy implementation of this too so that'll be neat.

This would then give a framework for loading anything else we need to create "quick fixes" for - most immediately data from the Krios, for example, which are simple formats but always long stacks... That would be for a separate PR though.

I didn't really want to touch this here but then thought it was a bit pointless creating lazy classes without lazy loading...